internal/core/cppapi_data.cpp

```diff
@@ -218,6 +216,7 @@ void fence(
         interpolation_method
     );
 
+    write_fillvalue(data.get(), noval_indicies, nsamples, fillValue);
```

Comment:
Wonder if we should also rename the third agrument here, in additional commit.
It is no longer "vertical_size", it gives no meaning here.

Reply:
sure